platform/platform-migratable/src/main/java/org/codice/ddf/platform/migratable/impl/PlatformMigratable.java

```diff
@@ -138,14 +162,90 @@ public void doImport(ImportMigrationContext context) {
         .stream()
         .map(context::getEntry)
         .forEach(me -> me.restore(false));
+    importKeystores(context);
+    importServiceWrapperFiles(context);
+  }
+
+  @Override
+  public void doVersionUpgradeImport(
+      ImportMigrationContext context,
+      java.util.Properties migrationProperties,
+      String migratableVersion) {
+    if (!VersionUtils.isValidMigratableVersion(context, migratableVersion, getVersion(), getId())) {
+      return;
+    }
+
+    LOGGER.debug("Upgrading system properties...");
+    try {
+      AccessUtils.doPrivileged(() -> upgradeCustomSystemProperties(context, migrationProperties));
+    } catch (IOException e) {
+      context.getReport().record(new MigrationException(CUSTOM_SYSTEM_PROPERTIES_ERROR, e));
+    }
+
+    LOGGER.debug("Importing remaining system files...");
+    PlatformMigratable.UPGRADEABLE_SYSTEM_PATHS
+        .stream()
+        .map(context::getEntry)
+        .forEach(me -> me.restore(false));
+    importKeystores(context);
+    importServiceWrapperFiles(context);
+  }
+
+  @VisibleForTesting
+  void upgradeCustomSystemProperties(
+      ImportMigrationContext context, java.util.Properties migrationProperties) throws IOException {
+    Properties importedProps = new Properties();
+    Properties currentProps;
+
+    File fileOnSystem =
+        Paths.get(System.getProperty("ddf.home"))
+            .toRealPath(LinkOption.NOFOLLOW_LINKS)
+            .resolve(Paths.get("etc", "custom.system.properties"))
+            .toFile();
+    currentProps = new Properties(fileOnSystem);
+
+    importedProps.load(
+        context
+            .getEntry(PlatformMigratable.CUSTOM_SYSTEM_PROPERTIES_PATH)
+            .getInputStream()
+            .orElseThrow(IOException::new));
+
+    List<String> systemPropertiesToPreserve =
+        migrationProperties.getProperty("system.properties.to.preserve") != null
```

Comment:
✏️ We should try to avoid hashing the properties object twice to get the property

Reply:
These Properties will no longer be passed in in the next commit, per our offline conversation. It'll just be a hardcoded list in the Java class instead.